spec/registry.xml

```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="registry.dtd" type="application/xml-dtd"?>
+<registry>
+  <pattern id="anyNumber" regex="-?[0-9]+(\.[0-9]+)"/>
+  <pattern id="positiveInteger" regex="[0-9]+"/>
+  <pattern id="currencyCode" regex="[A-Z]{3}"/>
+  <pattern id="timeZoneId" regex="[a-zA-Z/]+"/>
+  <pattern id="anythingNotEmpty" regex=".+"/>
+
+  <!-- I don't think that regexp is powerfull enough to
+    describe all things we might want, especially for input.
+    ISO 8601 is one example.
+    https://en.wikipedia.org/wiki/ISO_8601
+    So for now this is a very "sketchy" regexp.
+
+    One can also imagine phone numbers, emails, url, etc.
+
+    Even the regex for anyNumber (from the .md) is not OK:
+      .3  : Invalid (no digit before dot)
+      3   : Invalid (no dot + one digit after)
+            Brackets in regexp means grouping, not optional.
+      002 : Valid (OK to start with zero)
+      000 : Valid (OK to start with zero)
+  -->
+  <pattern id="iso8601" regex="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}\d{2}"/>
+
+  <function name="datetime">
+    <!-- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat -->
+    <description>Language-sensitive date and time formatting</description>
+
+    <formatSignature>
+      <input pattern="iso8601"/>
+      <!-- The predefined date formatting style to use. -->
+      <option name="dateStyle" values="full long medium short"/>
+      <!-- The predefined time formatting style to use. -->
+      <option name="timeStyle" values="full long medium short"/>
+      <!-- Calendar to use. -->
+      <option name="calendar" values="buddhist chinese coptic dangi ethioaa ethiopic gregory hebrew indian islamic islamic-umalqura islamic-tbla islamic-civil islamic-rgsa iso8601 japanese persian roc"/>
+      <!-- Numbering system to use. -->
+      <option name="numberingSystem" values="arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn limb mlym mong mymr orya tamldec telu thai tibt"/>
+      <!-- The time zone to use. The only value implementations must recognize
+         is "UTC"; the default is the runtime's default time zone.
+         Implementations may also recognize the time zone names of the IANA
+         time zone database, such as "Asia/Shanghai", "Asia/Kolkata",
+         "America/New_York".
+      -->
+      <option name="timeZone" pattern="timeZoneId"/>
```

Comment:
Don't forget offset time zones.

Reply:
It is 100% what ECMA says:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#timezone

I tried to not include anything that is not currently supported by ECMAScript, as we know there is strong opposition to that.

I'm for it, happy to discuss, but I didn't change this xml.

Added a topic in https://github.com/unicode-org/message-format-wg/issues/422